File: 001_Two_Sum/001.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        res = dict()
        for i, n in enumerate(nums):
            other = target - n
            if other in res:
                return [res[other], i]
            else:
                res[n] = i
        return []

File: 001_Two_Sum/001.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> m = new HashMap<>();
        for (int i = 0; i < nums.length; ++i) {
            int other = target - nums[i];
            if (m.containsKey(other))
                return new int[] {m.get(other), i};
            else
                m.put(nums[i], i);
        }
        return new int[] {};
    }
}